remote_test.go

```diff
@@ -493,3 +495,40 @@ func TestRemoteSSH(t *testing.T) {
 		t.Error("Expected remote heads")
 	}
 }
+
+func TestNewRemoteDetached(t *testing.T) {
+	_, err := NewRemoteDetached(TestGitRepoUri)
+	if err != nil {
+		t.Fatalf("error %v", err)
+	}
+}
+
+func TestRemoteDetached_LsDetached(t *testing.T) {
+	r, _ := NewRemoteDetached(TestGitRepoUri)
```

Comment:
```suggestion
	remote, err := NewRemoteDetached(TestGitRepoUri)
	checkFatal(t, err)
	defer remote.Free()
```

for consistency with other functions that create remotes (and to avoid memory leaks)